crates/bindings/src/table.rs

```diff
@@ -12,49 +12,45 @@ pub use spacetimedb_primitives::{ColId, IndexId};
 
 use crate::{bsatn, sys, Address, DeserializeOwned, Identity, IterBuf, Serialize, SpacetimeType, TableId};
 
-/// Implemented for every `TableHandle` struct generated in the client `module_bindings`
-/// and the module macroexpansion.
+/// Implemented for every `TableHandle` struct generated by the [`table`](macro@crate::table) macro.
+/// Contains methods that are present for every table, regardless of what unique constraints
+/// and indexes are present.
+///
+/// To get a `TableHandle`
+// TODO: should we rename this `TableHandle`? Documenting this, I think that's much clearer.
 pub trait Table: TableInternal {
     /// The type of rows stored in this table.
     type Row: SpacetimeType + Serialize + DeserializeOwned + Sized + 'static;
 
-    /// Returns the number of rows of this table in the TX state,
-    /// i.e. num(committed_state) + num(insert_table) - num(delete_table).
-    ///
-    /// This API is new to modules (though it previously existed in the Rust SDK)
-    /// and will require a new host function in the ABI.
+    /// Returns the number of rows of this table as of this transaction.
     fn count(&self) -> u64 {
         sys::datastore_table_row_count(Self::table_id()).expect("datastore_table_row_count() call failed")
     }
 
-    /// Iterate over all rows in the TX state,
-    /// i.e. committed_state ∪ insert_table ∖ delete_table.
+    /// Iterate over all rows of the table.
+    ///
+    /// For large tables, this can be a very slow operation!
+    /// Prefer [filtering](BTreeIndex::filter) a [`BTreeIndex`] or [finding](UniqueColumn::find) a [`UniqueColumn`] if
+    /// possible.
     #[inline]
     fn iter(&self) -> impl Iterator<Item = Self::Row> {
         let table_id = Self::table_id();
         let iter = sys::datastore_table_scan_bsatn(table_id).expect("datastore_table_scan_bsatn() call failed");
         TableIter::new(iter)
     }
 
-    /// Inserts `row` into the TX state,
-    /// i.e. removes it from the delete table or adds it to the insert table as appropriate.
+    /// Inserts `row` into the table.
     ///
     /// The return value is the inserted row, with any auto-incrementing columns replaced with computed values.
     /// The `insert` method always returns the inserted row,
     /// even when the table contains no auto-incrementing columns.
     ///
+    /// The returned row is a COPY of the row in the database.
```

Comment:
```suggestion
    /// The returned row is a copy of the row in the database.
```

I don't feel like we need capital letters for this.

Reply:
May be better to just splice this into the previous paragraph: "The return value is a copy of the inserted row, [...]."